sympy/physics/optics/utils.py

```diff
@@ -171,6 +175,67 @@ def refraction_angle(incident, medium1, medium2, normal=None, plane=None):
         return Ray3D(intersection_pt, direction_ratio=drs)
 
 
+def fresnel_coefficients(angle_of_incidence, medium1, medium2):
+    """
+    This function uses Fresnel equations to calculate reflection and
+    transmission coefficients. Those are obtained for both polarisations
+    when the electric field vector is in the plane of incidence (labelled 'p')
+    and when the electric field vector is perpendicular to the plane of
+    incidence (labelled 's'). There four real coefficients unless the incident
+    ray reflects in total internal in which case there are two complex ones.
+    Angle of incidence is the angle between the incident ray and the surface
+    normal. ``medium1`` and ``medium2`` can be ``Medium`` or any sympifiable object.
+
+    Parameters
+    ==========
+
+    angle_of_incidence : sympifiable
+    medium1 : Medium or sympifiable
+        Medium 1 or its refractive index
+    medium2 : Medium or sympifiable
+        Medium 2 or its refractive index
+
+    Examples
+    ========
+
+    >>> from sympy.physics.optics import fresnel_coefficients
+    >>> fresnel_coefficients(0.3, 1, 2)
+    [0.317843553417859, -0.348645229818821,
+            0.658921776708929, 0.651354770181179]
+    >>> fresnel_coefficients(0.6, 2, 1)
+    [-0.235625382192159 - 0.971843958291041*I,
+             0.816477005968898 - 0.577377951366403*I]
+    """
+
+    if isinstance(medium1, Medium):
+        n1 = medium1.refractive_index
+    else:
+        n1 = sympify(medium1)
+
+    if isinstance(medium2, Medium):
+        n2 = medium2.refractive_index
+    else:
+        n2 = sympify(medium2)
+
+    angle_of_refraction = asin(n1*sin(angle_of_incidence)/n2)
+    try:
+        angle_of_total_internal_reflection_onset = critical_angle(n1, n2)
+    except ValueError:
+        angle_of_total_internal_reflection_onset = None
+
+    if angle_of_total_internal_reflection_onset == None or angle_of_total_internal_reflection_onset > angle_of_incidence:
+        R_s = -sin(angle_of_incidence-angle_of_refraction)/sin(angle_of_incidence+angle_of_refraction)
+        R_p = tan(angle_of_incidence-angle_of_refraction)/tan(angle_of_incidence+angle_of_refraction)
+        T_s = 2*sin(angle_of_refraction)*cos(angle_of_incidence)/sin(angle_of_incidence+angle_of_refraction)
+        T_p = 2*sin(angle_of_refraction)*cos(angle_of_incidence)/(sin(angle_of_incidence+angle_of_refraction)*cos(angle_of_incidence-angle_of_refraction))
+        return [R_p, R_s, T_p, T_s]
+    else:
+        n = n2/n1
+        R_s = simplify((cos(angle_of_incidence)-I*sqrt(sin(angle_of_incidence)**2-n**2))/(cos(angle_of_incidence)+I*sqrt(sin(angle_of_incidence)**2-n**2)))
+        R_p = simplify((n**2*cos(angle_of_incidence)-I*sqrt(sin(angle_of_incidence)**2-n**2))/(n**2*cos(angle_of_incidence)+I*sqrt(sin(angle_of_incidence)**2-n**2)))
```

Comment:
These lines could possibly be broken down as they are longer than 79 characters. Adding spaces between individual terms separated by operators (like `+`, `-`) will improve the readability of expression.

Is `simplify` necessary for this? It is generally preferable to use only specific simplifications (more information at https://docs.sympy.org/latest/tutorial/simplification.html) in library code, as `simplify` can be slow.